Framework/src/KafkaPoller.cxx

```diff
@@ -0,0 +1,166 @@
+  std::string_view state;
+  events::OpStatus transitionStatus;
+  std::string_view environmentID;
+  int runNumber;
+};
+
+bool operator==(const events::Ev_RunEvent& runEvent, const Ev_RunEventPartial& runEvenPartial)
+{
+  // TODO: Should we check whether the error is empty?
+  if (runEvent.state() != runEvenPartial.state ||
+      runEvent.transitionstatus() != runEvenPartial.transitionStatus) {
+    return false;
+  }
+
+  if (isRunNumberSet(runEvenPartial.runNumber) && runEvent.runnumber() != runEvenPartial.runNumber) {
+    return false;
+  }
+
+  if (isEnvironmentIdSet(runEvenPartial.environmentID) && runEvent.environmentid() != runEvenPartial.environmentID) {
+    return false;
+  }
+
+  return true;
+}
+
+auto recordToEvent(const kafka::Value& kafkaRecord) -> std::optional<events::Event>
+{
+  events::Event event;
+
+  if (!event.ParseFromArray(kafkaRecord.data(), kafkaRecord.size())) {
+    ILOG(Error, Ops) << "Received wrong or inconsistent data while parser Event from kafka proto" << ENDM;
+    return std::nullopt;
+  }
+
+  return event;
+}
+
+void fillActivityWithoutTimestamp(const events::Event& event, Activity& activity)
+{
+  if (event.has_runevent()) {
+    auto& runEvent = event.runevent();
+
+    if (!isRunNumberSet(activity.mId)) {
+      activity.mId = runEvent.runnumber();
+    }
+
+    if (!isEnvironmentIdSet(activity.mProvenance)) {
+      activity.mProvenance = runEvent.environmentid();
+    }
+  }
+}
+
+void start_of_run::fillActivity(const events::Event& event, Activity& activity)
+{
+  fillActivityWithoutTimestamp(event, activity);
+  activity.mValidity.setMin(event.timestamp());
+}
+
+bool start_of_run::check(const events::Event& event, const std::string& environmentID, int runNumber)
+{
+  if (!event.has_runevent()) {
+    return false;
+  }
+
+  const auto& runEvent = event.runevent();
+
+  if (runEvent.transition() != "START_ACTIVITY") {
+    return false;
+  }
+
+  return runEvent == Ev_RunEventPartial{ "CONFIGURED", events::OpStatus::STARTED, environmentID, runNumber };
+}
+
+void end_of_run::fillActivity(const events::Event& event, Activity& activity)
+{
+  fillActivityWithoutTimestamp(event, activity);
+  activity.mValidity.setMax(event.timestamp());
+}
+
+bool end_of_run::check(const events::Event& event, const std::string& environmentID, int runNumber)
+{
+  if (!event.has_runevent()) {
+    return false;
+  }
+
+  const auto& runEvent = event.runevent();
+
+  if (runEvent.transition() != "STOP_ACTIVITY" && runEvent.transition() != "TEARDOWN") {
+    return false;
+  }
+
+  return runEvent == Ev_RunEventPartial{ "RUNNING", events::OpStatus::STARTED, environmentID, runNumber };
+}
+
+} // namespace proto
+
+kafka::Properties createProperties(const std::string& brokers, const std::string& groupId)
+{
+  return { { { "bootstrap.servers", { brokers } },
+             { "group.id", { groupId } },
+             { "enable.auto.commit", { "true" } },
+             { "auto.offset.reset", { "latest" } } } };
```

Comment:
does `auto.offset.reset` == `latest` mean that we will consume only the latest event of a kind? What if two SORs happen in between two consecutive postprocessing cycles (two `tryTrigger` calls)?

Reply:
it means, that we when new consumer with new group id is connected to cluster it will consume only latest messages, if some consumer connects to already existing group, than it will resume from the offset in the group.

Reply:
ah, all good then, I think this is what we need